### PR TITLE
Added info modal in meshery

### LINKED ISSUE #8493: [UI] Expose Catalog Request Data
### Current Behavior

Currently, catalog request data is not exposed in Meshery UI. It's only shown when a user fills the form to publish.

Screenshot 2023-08-16 at 9 27 14 PM

### Desired Behavior

Ensure that Catalog request data is exposed in readonly format on the back of the configuration card.

### Mockups

---

### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)

- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🖥 [Contributing to Meshery UI](https://docs.meshery.io/project/contributing/contributing-ui)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by Yashsharma1911**
**Notes for Reviewers**

This PR fixes #8493 

This PR adds new info modal in Pattern and Filters page to expose catalog data so that users can get information of design and filter and can edit at the same time if they are owner of that design or pattern

This information will be mostly same as when you publish a design

Screenshot

Screenshot 2023-09-27 at 2 37 46 AM

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by Yashsharma1911** [comment]
Task linked: [CU-85zttdnfm [UI] Expose Catalog Request Data](https://app.clickup.com/t/85zttdnfm)

**Comment by leecalcote** [comment]
@Yashsharma1911 still WIP or ready for review?

**Comment by saurabh100ni** [comment]
Let's discuss this on the Meshery dev call tomorrow,
Please add this as your agenda item in Docs: https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit

**Comment by Yashsharma1911** [comment]
https://github.com/meshery/meshery/assets/71271069/6878c16b-407b-4b5e-b0db-05fc9157d536

File: ui/components/Modals/Information/infoModal.js [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+const PATTERN_PLURAL = 'patterns';
+
+const InfoModal = (props) => {
+  const {
+    infoModalOpen,
+    handleInfoModalClose,
+    resourceOwnerID, //id to get data of user to get name of that person
+    dataName,
+    selectedResource,
+    currentUserID, // id of current user
```
Comment (Yashsharma1911) [inline ui/components/Modals/Information/infoModal.js:50]:
```suggestion
    currentUserID,
```

File: ui/components/Modals/Information/infoModal.js [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+
+const APPLICATION_PLURAL = 'applications';
+const FILTER_PLURAL = 'filters';
+const PATTERN_PLURAL = 'patterns';
+
+const InfoModal = (props) => {
+  const {
+    infoModalOpen,
+    handleInfoModalClose,
+    resourceOwnerID, //id to get data of user to get name of that person
```
Comment (Yashsharma1911) [inline ui/components/Modals/Information/infoModal.js:47]:
```suggestion
    resourceOwnerID,
```

File: ui/components/Modals/Information/infoModal.js [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+    dataName,
+    selectedResource,
+    currentUserID, // id of current user
+    formSchema,
+  } = props;
+
+  // const theme = uiTheme.theme;
```
Comment (Yashsharma1911) [inline ui/components/Modals/Information/infoModal.js:54]:
```suggestion
```

File: ui/components/Modals/Information/infoModal.js [hunk 4]
```diff
@@ -0,0 +1,365 @@
...
+        body = {
+          body: JSON.stringify({
+            pattern_data: {
+              catalog_data: formState,
+              id: selectedResource.id,
+            },
+          }),
+        };
+      } else if (dataName === FILTER_PLURAL) {
+        setSaveFormLoading(true);
```
Comment (Yashsharma1911) [inline ui/components/Modals/Information/infoModal.js:92]:
```suggestion
```

File: ui/components/Modals/Information/infoModal.js [hunk 5]
```diff
@@ -0,0 +1,365 @@
...
+
+      axios
+        .post(dataName === PATTERN_PLURAL ? PATTERN_ENDPOINT : FILTER_ENDPOINT, body)
+        .then(() => {
+          setSaveFormLoading(false);
+          notify({
+            message: `${selectedResource.name} data saved successfully`,
+            event_type: EVENT_TYPES.SUCCESS,
+          });
+        })
```
Comment (Yashsharma1911) [inline ui/components/Modals/Information/infoModal.js:111]: @theBeginner86 this is how I'm sending catalog data

Comment (theBeginner86) [reply ui/components/Modals/Information/infoModal.js]: Looking into it now....

**Comment by Yashsharma1911** [comment]
@theBeginner86 thanks more making changes let's merge this if it is working

**Comment by theBeginner86** [comment]
> @theBeginner86 thanks more making changes let's merge this if it is working

yes, I was just tying this off. Would merge as soon as I'm done...

Thanks for confirming the status from your side 👍

**Comment by theBeginner86** [comment]
We are ready now, @Yashsharma1911

**Pull request merged**